### ISSUE TITLE
Ribasim Python: stop exposing `Ribasim.Node` and node IDs to users

### ISSUE THREAD

**Issue by Hofer-Julian** [Deltares/Ribasim#252]
Thanks to @SouthEndMusic's work, users will soon have fewer chances to mess up their model input when using Ribasim Python.
https://github.com/Deltares/Ribasim/issues/245

What would be even better, if users wouldn't have to think about node IDs at all. They would create specific node types with the corresponding geometries attached. When new nodes are created that way, they would access a singleton allocator which returns new and unique node IDs.

For creating edges, the interfaces might change like this:

```diff
- from_id = np.array(
-    [1, 2], dtype=np.int64
-)
- to_id = np.array(
-    [2, 3], dtype=np.int64
-)

+ from_id = [basin.nodes[0], linear_resistance.nodes[0]]
+ to_id = [linear_resisitance.nodes[0], basin.nodes[1]]
```

**Comment by visr** [comment]
#446 introduces a `get_node_ids_and_types` function that addresses a part of this. This probably needs to be removed/refined later. Right now it is guessing available tables for node types, `["static", "time", "condition"]`.

https://github.com/Deltares/Ribasim/pull/446/files#diff-1781035826cbdb5bd206a7c07398ee9d29ad5c34fa4312b6a0b07db7e154d3ab

**Comment by visr** [comment]
Fixing this would probably also fix #212, since you would not pass the Node table anymore so there is no way it cannot match.
